src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIListDAO.java

```diff
@@ -196,4 +210,47 @@ public List<BrAPIListSummary> createBrAPILists(List<BrAPIListNewRequest> brapiLi
 
         throw new ApiException("No response after creating list");
     }
+
+    public void deleteBrAPIList(String listDbId, UUID programId, boolean hardDelete) throws ApiException {
+        // TODO: Switch to using the ListsApi from the BrAPI client library once the delete endpoints are merged into it
+        var programBrAPIBaseUrl = getProgramBrAPIBaseUrl(programId);
+        var requestUrl = HttpUrl.parse(programBrAPIBaseUrl + "/lists/" + listDbId).newBuilder();
+        requestUrl.addQueryParameter("hardDelete", Boolean.toString(hardDelete));
+        HttpUrl url = requestUrl.build();
+        var brapiRequest = new Request.Builder().url(url)
+                .method("DELETE", null)
+                .addHeader("Content-Type", "application/json")
+                .build();
+
+        makeCall(brapiRequest);
+    }
+
+    private void makeCall(Request brapiRequest) throws ApiException {
+        OkHttpClient client = new OkHttpClient.Builder()
+                .readTimeout(5, TimeUnit.MINUTES)
+                .build();
+        try {
+            client.newCall(brapiRequest).execute();
+        } catch (IOException e) {
+            log.error("Error calling BrAPI Service", e);
+            throw new ApiException("Error calling BrAPI Service");
+        }
+    }
```

Comment:
@dmeidlin could you make `makeCall` a static method in a utility class or something like that so that all the DAOs that issue similar delete requests can reuse it?

Reply:
@mlm483 You'll want to merge in the latest version of that method. I modified it to handle exceptions like a 404 response from the brapi server.